File: backend/app/schemas/trade.py
"""Trade schemas."""

from datetime import datetime
from decimal import Decimal
from uuid import UUID

from pydantic import BaseModel


class TradeBase(BaseModel):
    """Base trade schema."""

    symbol: str
    side: str
    order_type: str = "market"
    quantity: Decimal
    price: Decimal


class TradeCreate(TradeBase):
    """Trade creation schema."""

    portfolio_id: UUID
    strategy_id: UUID | None = None


class TradeResponse(TradeBase):
    """Trade response schema."""

    id: UUID
    portfolio_id: UUID
    strategy_id: UUID | None = None
    filled_quantity: Decimal | None = None
    filled_price: Decimal | None = None
    commission: Decimal
    status: str
    exchange_order_id: str | None = None
    pnl: Decimal | None = None
    notes: str | None = None
    executed_at: datetime | None = None
    created_at: datetime
    updated_at: datetime

    model_config = {"from_attributes": True}

File: frontend/src/types/trade.ts
export type TradeType = 'market' | 'limit' | 'stop' | 'stop_limit'
export type TradeSide = 'buy' | 'sell'
export type TradeStatus = 'pending' | 'filled' | 'partial' | 'cancelled' | 'rejected'

export interface Trade {
  id: string
  portfolioId: string
  strategyId?: string
  symbol: string
  side: TradeSide
  type: TradeType
  status: TradeStatus
  quantity: number
  filledQuantity: number
  price?: number
  filledPrice?: number
  stopPrice?: number
  commission: number
  pnl?: number
  createdAt: string
  filledAt?: string
}

export interface TradeHistory {
  trades: Trade[]
  totalPnl: number
  winCount: number
  lossCount: number
  winRate: number
}

export interface BacktestResult {
  id: string
  strategyId: string
  startDate: string
  endDate: string
  initialCapital: number
  finalCapital: number
  totalReturn: number
  totalReturnPercent: number
  maxDrawdown: number
  maxDrawdownPercent: number
  sharpeRatio: number
  sortinoRatio: number
  winRate: number
  profitFactor: number
  totalTrades: number
  winningTrades: number
  losingTrades: number
  averageWin: number
  averageLoss: number
  equityCurve: EquityPoint[]
  trades: Trade[]
  createdAt: string
}

export interface EquityPoint {
  date: string
  equity: number
  drawdown: number
}
